dot/parachain/prospective-parachains/errors.go

```diff
@@ -0,0 +1,81 @@
+package prospectiveparachains
+
+import (
+	"errors"
+	"fmt"
+
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+var (
+	ErrCandidateAlreadyKnown           = errors.New("candidate already known")
+	ErrZeroLengthCycle                 = errors.New("candidate's parent head is equal to its output head. Would introduce a cycle") //nolint:lll
+	ErrCycle                           = errors.New("candidate would introduce a cycle")
+	ErrMultiplePaths                   = errors.New("candidate would introduce two paths to the same output state")
+	ErrIntroduceBackedCandidate        = errors.New("attempting to directly introduce a Backed candidate. It should first be introduced as Seconded") //nolint:lll
+	ErrParentCandidateNotFound         = errors.New("could not find parent of the candidate")
+	ErrRelayParentMovedBackwards       = errors.New("relay parent would move backwards from the latest candidate in the chain")     //nolint:lll
+	ErrPersistedValidationDataMismatch = errors.New("candidate does not match the persisted validation data provided alongside it") //nolint:lll
```

Comment:
Do we need all these exported?
Also the same question for other types you have added in this file

Reply:
they don't need to be exported, we use them as sentinel errors which helps us to contextualize when an error occurs.